{{cookiecutter.repo_name}}/.github/actions/get-stable-python-version/get_python_version.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+
+# MIT License
+
+# Copyright (c) 2022 MDAnalysis Development Team
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+"""
+Script to grab the latest N version of Python.
+This is very helpful for CI purposes.
+Note that only final releases are considered.
+
+Example usage, when 3.10 is the latest active release, and 3.11 is in beta:
+
+$ ./get_python_version.py
+3.10
+$ ./get_python_version.py --field major
+3
+$ ./get_python_version.py --field micro
+3.10.6
+$ ./get_python_version.py --field minor --last-n-version 1
+3.9
+$ ./get_python_version.py --field minor --last-n-version 2
+3.8
+$ ./get_python_version.py --field micro --last-n-version 2
+3.10.4
+
+"""
+
+
+import argparse
+import urllib.request
+import json
+from typing import NamedTuple, List, Tuple
+
+FIELD_CHOICES = ["major", "minor", "micro"]
+
+parser = argparse.ArgumentParser(
+    description="Get the last n Python versions",
+)
+parser.add_argument(
+    "--last-n-version",
+    type=int,
+    help=(
+        "The version to check. 0 returns the latest release; "
+        "1 returns the one before that"
+    ),
+    default=0,
+)
+parser.add_argument(
+    "--field",
+    choices=FIELD_CHOICES,
+    help="Field to select last-n-version on",
+    default="minor",
+)
+
+
+class VersionInfo(NamedTuple):
```

Comment:
Isn't there a whole version package maintained by pypa that does this?

Reply:
Is it in the standard library? I couldn't find anything from a quick skim 😅 the script needs to be able to run on the base Python installs on the GH runners

Reply:
maybe we can vendor version.py?

Reply:
(my viewpoint here being that maintaining a fully compliant version string reader is quite the pain iirc and `packaging`'s version.py is sorta made to be vendored for these purposes iirc)

Reply:
actually maybe just ignore me here, that'd be a lot more complexity for this task